lib/home-manager-configuration.nix

```diff
@@ -0,0 +1,21 @@
+{ self, lib, home-manager }:
+
+{ configuration
+, system
+, homeDirectory
+, username
+, modules ? [ ]
```

Comment:
this doesn't match what's done in #32. Let's be consistent in the way we name things, even if that means some extra code on our side.

Reply:
You mean global and hm? hmm do you think we will ever have two+ systems in home-manager to merit a split?

Reply:
We will probably never have something else than home-manager in home-manager configurations, but what if I want to do something like this in my soxincfg:

```
hmModules = [ /* ... */ ];
nixosHosts = [ "my_host" ];
hmHosts = [ "my_other_host" ];

nixosConfigurations = /* some method like genAttrs */ soxin.nixosConfiguration {
  /* ... */
  inherit hmModules;
}:

homeManagerConfigurations = /* some method like genAttrs */ soxin.homeManagerConfiguration {
  /* ... */
  inherit hmModules;
};
```

Reply:
sure, I'm fine with the name, let me change it now.

Reply:
done